Lib/email/_msgid.py

```diff
@@ -0,0 +1,29 @@
+# This function used to be part of email.utils,
+# but has been separated out to speedup the import of that module
+
+import os
+import random
+import socket
+import time
+
+def make_msgid(idstring=None, domain=None):
+    """Returns a string suitable for RFC 2822 compliant Message-ID, e.g:
+
+    <142480216486.20800.16526388040877946887@nightshade.la.mastaler.com>
+
+    Optional idstring if given is a string used to strengthen the
+    uniqueness of the message id.  Optional domain if given provides the
+    portion of the message id after the '@'.  It defaults to the locally
+    defined hostname.
+    """
+    timeval = int(time.time()*100)
+    pid = os.getpid()
+    randint = random.getrandbits(64)
+    if idstring is None:
+        idstring = ''
+    else:
+        idstring = '.' + idstring
+    if domain is None:
+        domain = socket.getfqdn()
+    msgid = '<%d.%d.%d%s@%s>' % (timeval, pid, randint, idstring, domain)
```

Comment:
Appreciate this is import time not general refactoring -- but we could consider making this an f-string!
```suggestion
    msgid = f'<{timeval}.{pid}.{randint}{idstring}@{domain}>'
```

Reply:
I'd rather refrain from (what could be perceived as) gratuitous refactoring here, since I'm not the primary maintainer of this module — I'll defer to the `email` maintainers on what they'd like here 👍